src/Http/Http.Results/src/ObjectResult.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Http.Result
+{
+    internal partial class ObjectResult : IResult
+    {
+        /// <summary>
+        /// Creates a new <see cref="ObjectResult"/> instance with the provided <paramref name="value"/>.
+        /// </summary>
+        public ObjectResult(object? value, int statusCode)
+        {
+            Value = value;
+            StatusCode = statusCode;
+        }
+
+        /// <summary>
+        /// The object result.
+        /// </summary>
+        public object? Value { get; }
+
+        /// <summary>
+        /// Gets or sets the HTTP status code.
+        /// </summary>
+        public int? StatusCode { get; }
+
+        public Task ExecuteAsync(HttpContext httpContext)
+        {
+            var loggerFactory = httpContext.RequestServices.GetRequiredService<ILoggerFactory>();
```

Comment:
We're going to want to cache these calls to the logger, but that should be in another PR. My guess is that we haven't looked at this cost in MVC.